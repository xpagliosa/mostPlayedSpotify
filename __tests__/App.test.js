import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import App from '../src';

describe('Snapshots App Component', () => {
  let wrapper = shallow(
    <App 
      loading={true}
      loadingText="Carregando Spotify..."
      tryAgain={true}
      artistID="6jesZl7rJFJZ8d5i9pX5q8"
    />
  );
  it('should renders Loading e TryAgain correctly', () => {
    expect(wrapper).toMatchSnapshot();
  }),

  it('shouls render Loading correctly', () => {
    wrapper.setState({ tryAgain: false });
    expect(wrapper).toMatchSnapshot();
  }),

  it('shouls render ID Tagore correctly', () => {
    wrapper.setState({ loading: false });
    expect(wrapper).toMatchSnapshot();
  }),
    
  it('shouls render ID Confraria correctly', () => {
    wrapper.setState({ artistID: "5slvIMrXIIfl1kNYszWP5G" });
    expect(wrapper).toMatchSnapshot();
  });
})
