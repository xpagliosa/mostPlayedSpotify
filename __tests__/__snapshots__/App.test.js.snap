// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots App Component should renders Loading e TryAgain correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    artistID="6jesZl7rJFJZ8d5i9pX5q8"
    loading={true}
    loadingText="Carregando Spotify..."
    tryAgain={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          barStyle="light-content"
          showHideTransition="fade"
        />,
        <View
          style={
            Object {
              "height": 30,
              "paddingTop": 20,
            }
          }
        />,
        <View
          style={
            Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#333",
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "barStyle": "light-content",
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "height": 30,
            "paddingTop": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />,
          "style": Object {
            "backgroundColor": "#333",
            "flex": 1,
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "loadingText": "",
            "signIn": [Function],
            "tryAgain": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            barStyle="light-content"
            showHideTransition="fade"
          />,
          <View
            style={
              Object {
                "height": 30,
                "paddingTop": 20,
              }
            }
          />,
          <View
            style={
              Object {
                "backgroundColor": "#333",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#333",
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "barStyle": "light-content",
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "height": 30,
              "paddingTop": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />,
            "style": Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "loadingText": "",
              "signIn": [Function],
              "tryAgain": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Snapshots App Component shouls render ID Confraria correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    artistID="5slvIMrXIIfl1kNYszWP5G"
    loading={false}
    loadingText="Carregando Spotify..."
    tryAgain={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          barStyle="light-content"
          showHideTransition="fade"
        />,
        <View
          style={
            Object {
              "height": 30,
              "paddingTop": 20,
            }
          }
        />,
        <View
          style={
            Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#333",
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "barStyle": "light-content",
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "height": 30,
            "paddingTop": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />,
          "style": Object {
            "backgroundColor": "#333",
            "flex": 1,
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "loadingText": "",
            "signIn": [Function],
            "tryAgain": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            barStyle="light-content"
            showHideTransition="fade"
          />,
          <View
            style={
              Object {
                "height": 30,
                "paddingTop": 20,
              }
            }
          />,
          <View
            style={
              Object {
                "backgroundColor": "#333",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#333",
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "barStyle": "light-content",
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "height": 30,
              "paddingTop": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />,
            "style": Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "loadingText": "",
              "signIn": [Function],
              "tryAgain": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Snapshots App Component shouls render ID Tagore correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    artistID="6jesZl7rJFJZ8d5i9pX5q8"
    loading={false}
    loadingText="Carregando Spotify..."
    tryAgain={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          barStyle="light-content"
          showHideTransition="fade"
        />,
        <View
          style={
            Object {
              "height": 30,
              "paddingTop": 20,
            }
          }
        />,
        <View
          style={
            Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#333",
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "barStyle": "light-content",
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "height": 30,
            "paddingTop": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />,
          "style": Object {
            "backgroundColor": "#333",
            "flex": 1,
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "loadingText": "",
            "signIn": [Function],
            "tryAgain": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            barStyle="light-content"
            showHideTransition="fade"
          />,
          <View
            style={
              Object {
                "height": 30,
                "paddingTop": 20,
              }
            }
          />,
          <View
            style={
              Object {
                "backgroundColor": "#333",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#333",
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "barStyle": "light-content",
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "height": 30,
              "paddingTop": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />,
            "style": Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "loadingText": "",
              "signIn": [Function],
              "tryAgain": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Snapshots App Component shouls render Loading correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    artistID="6jesZl7rJFJZ8d5i9pX5q8"
    loading={true}
    loadingText="Carregando Spotify..."
    tryAgain={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          barStyle="light-content"
          showHideTransition="fade"
        />,
        <View
          style={
            Object {
              "height": 30,
              "paddingTop": 20,
            }
          }
        />,
        <View
          style={
            Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#333",
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "barStyle": "light-content",
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "height": 30,
            "paddingTop": 20,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Loading
            loadingText=""
            signIn={[Function]}
            tryAgain={false}
          />,
          "style": Object {
            "backgroundColor": "#333",
            "flex": 1,
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "loadingText": "",
            "signIn": [Function],
            "tryAgain": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            barStyle="light-content"
            showHideTransition="fade"
          />,
          <View
            style={
              Object {
                "height": 30,
                "paddingTop": 20,
              }
            }
          />,
          <View
            style={
              Object {
                "backgroundColor": "#333",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#333",
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "barStyle": "light-content",
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "height": 30,
              "paddingTop": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Loading
              loadingText=""
              signIn={[Function]}
              tryAgain={false}
            />,
            "style": Object {
              "backgroundColor": "#333",
              "flex": 1,
              "flexDirection": "column",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "loadingText": "",
              "signIn": [Function],
              "tryAgain": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
